estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,63 @@
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+
+class PropertyOfferModel(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property Offer model"
+    _order = "price desc"
+
+    price = fields.Float()
+    status = fields.Selection(
+        selection=[
+            ("accepted", "Accepted"),
+            ("refused", "Refused")
+        ],
+        copy=False
+    )
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+    property_type_id = fields.Many2one(related="property_id.property_type_id", store=True)
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_deadline", inverse="_inverse_deadline")
+
+    _check_price = models.Constraint(
+        "CHECK(price >= 0)",
+        "The price of the offer must be positive."
+    )
+
+    @api.depends("validity")
+    def _compute_deadline(self):
+        for record in self:
+            record.date_deadline = fields.Date.add((record.create_date or fields.Datetime.now()), days=record.validity)
+
+    def _inverse_deadline(self):
+        for record in self:
+            record.validity = (record.date_deadline - fields.Date.to_date(record.create_date)).days if record.date_deadline else record.validity
+
+    @api.model
+    def create(self, vals_list: list[dict]):
+        for val in vals_list:
+            estate_property = self.env["estate.property"].browse(val["property_id"])
```

Comment:
You shouldn't do sql request inside a loop 😃. It will do a lot of requests to your db as the create method can be used to create multiple records. Get first all the needed ids and then make a request with all of them with_prefetch(), which will allow you to get records by their id with browse inside the loop without doing any requests to the data base.  Here also you can see that I used EstateProperties instead of estate_properties, that's an Odoo convention. When a variable represent a model we have to use the same convention than for the name of a python class. Sorry for that one, I have been a little hard on that one 😅. Just remember no SQL queries inside loops.
```suggestion
       EstateProperties = self.env["estate.property"].with_prefetch(
            [vals["property_id"] for vals in vals_list]
        )
        for val in vals_list:
            estate_property = EstateProperties.browse(val["property_id"])
```